gcsio/src/main/java/com/google/cloud/hadoop/gcsio/GoogleCloudStorageGrpcReadChannel.java

```diff
@@ -419,6 +422,19 @@ public int read(ByteBuffer byteBuffer) throws IOException {
     }
 
     long effectivePosition = positionInGrpcStream + bytesToSkipBeforeReading;
+
+    /* resIterator is null on the first read (position = 0) or when a seek is performed (and when
+      there are exceptions). So if footerBuffer is null and we are trying to read into footer
+      region, we may just cache the footer
+    */
+    if ((resIterator == null)
+        && (footerBuffer == null)
+        && (effectivePosition >= footerStartOffsetInBytes)) {
```

Comment:
Please add a test to verify the behaviour of partially overlapping footer content, which induces an range read call

Reply:
Added 2 more test cases.